usl_pipeline/cloud_functions/main.py

```diff
@@ -143,30 +143,53 @@ def wrapper(cloud_event: functions_framework.CloudEvent) -> None:
 def build_and_upload_study_area_chunk(
     cloud_event: functions_framework.CloudEvent,
 ) -> None:
-    """Creates a study area chunk and uploads it to GCS.
+    """Creates a study area chunk, uploads to bucket, and writes to metastore.
 
     This function is triggered when files containing raw geo data for a study area
-    are uploaded to GCS. It will group the files into TAR-files and upload it to a
-    bucket.
+    are uploaded to GCS. It will load the file into study area chunks bucket and
+    write a new study area to metastore if it does not exist.
     """
-    file_name = cloud_event.data["name"]
+    file_name = pathlib.PurePosixPath(cloud_event.data["name"])
     bucket_name = cloud_event.data["bucket"]
 
-    if re.search(file_names.WPS_DOMAIN3_NC_REGEX, file_name):
+    # For AtmoML, only 3rd domain (500m) output files will be used
+    if re.search(file_names.WPS_DOMAIN3_NC_REGEX, file_name.name):
         storage_client = storage.Client()
+        db = firestore.Client()
+
         bucket = storage_client.bucket(bucket_name)
-        file_blob = bucket.blob(file_name)
+        file_blob = bucket.blob(str(file_name))
 
         # For WRF, we can treat each snapshot output file as its own chunk. If
         # multiple snapshots must be processed together, we can choose to group
         # the files together in a TAR before uploading.
-        with file_blob.open("rb") as f:
+        with file_blob.open("rb") as fd:
             study_area_chunk_bucket = storage_client.bucket(
                 cloud_storage.STUDY_AREA_CHUNKS_BUCKET
             )
-            study_area_chunk_bucket.blob(file_name).upload_from_file(f)
+            study_area_chunk_bucket.blob(str(file_name)).upload_from_file(fd)
 
-        # TODO: Write to metastore - create new chunk entry
+            # File names should be in the form <study_area_name>/<file_name>
+            study_area_name = file_name.parent.name
+
+            # Write metastore entry for study area if it does not exist
+            study_area_exists = (
+                metastore.StudyArea.get_ref(db, str(study_area_name)).get().exists
+            )
+            if not study_area_exists:
+                with xarray.open_dataset(fd) as ds:
+                    x_ll_meters = ds.data_vars["XLONG_M"].values[0, 0, 0] * 111
```

Comment:
Why times 111 here? A comment would be helpful here.

Reply:
This is still WIP. The rule of thumb for converting meters to degrees is to use this conversion: 111kilometer = 1 degree. But this is only a rough estimate and changes with latitude. I wouldn't use this tbh. It'll be helpful to know how x_ll and y_ll are mapped out in FloodML

Reply:
it's part of the input data for flood -- we have a .tiff with x_ll & y_ll & CRS as an input to define the raster

Reply:
the x_ll & y_ll are ultimately what we'll send to ClimaSens to build the dashboards, so it will need to be accurate!

Reply:
for Manhattan we have
```
cell_size: 2
col_count: 5835
crs: "EPSG:32618"
row_count: 10945
x_ll_corner: 580459.5669010617
y_ll_corner: 4504014.133973204
```

Reply:
This is really helpful. cell_size in that case should be 500 (I would remove the meter squared comment -> its misleading)
col_count and row_count will be 200 for all cities
I need to do some digging to be sure about the crs and the x_ll and y_ll

Reply:
Thanks @waltaskew @MadhaviJain90 - updated. I still have an open about CRS tho - @MadhaviJain90 how do we find out what system we use for WRF?

Reply:
At a glance, looks like the CRS can be found using this:
https://wrf-python.readthedocs.io/en/latest/user_api/generated/wrf.get_cartopy.html#wrf-get-cartopy

x_ll_corner and y_ll_corner are the coordinates of the lower left corner of the lower left cell. They don't need to be converted into meters by multiplying by 111. Keeping them as degrees is fine.

Reply:
Also, want to confirm (0, 0) is the lower left corner and not (0, 200) -- can double check the lat lon values to see where the mins are

Reply:
So at the end of the day, we're going to use the CRS to pull the predictions into geopandas here:
https://github.com/UrbanSystemsLab/climateiq-frontend/blob/2f54adf0615ee0a9d170b27550ede602972361c9/cloud_functions/climateiq_data_export_cf/main.py#L260
and then export them to ClimaSens.
Can you confirm that geopandas is happy taking in a value like 'lambert' and such? Or if we need to supply a value like 'EPSG:9802' instead? (Could just hop into the unit tests for the above code and swap 'lambert' in as the CRS and see what happens to test this quickly.)

Reply:
No it look slike geopands will expect an authority string: https://geopandas.org/en/stable/docs/reference/api/geopandas.GeoDataFrame.html

I've added the code to get the source crs from the wps files though. pleas take a looK!